nexus_constructor/json/json_warnings.py

```diff
@@ -30,5 +40,88 @@ class NXClassAttributeMissing:
 
 
 JsonWarning = Union[
-    InvalidJson, TransformDependencyMissing, NameFieldMissing, NXClassAttributeMissing
+    InvalidJson,
+    InvalidShape,
+    InvalidTransformation,
+    TransformDependencyMissing,
+    NameFieldMissing,
+    NXClassAttributeMissing,
 ]
+
+_T = TypeVar("_T", JsonWarning, "JsonWarningsContainer")
+
+
+class JsonWarningsContainer(List[JsonWarning]):
```

Comment:
Just for curiosity, Why `List[JsonWarning] `and not just `List`?

Reply:
As we discussed in chat, in principle it is OK to use List but it is mostly to be super clear that it is a JsonWarning container. :)